vendor/lunasvg/source/lunasvg.cpp

```diff
@@ -195,7 +195,9 @@ void Document::render(Bitmap bitmap, const Matrix& matrix, std::uint32_t backgro
     state.transform = Transform(matrix.a, matrix.b, matrix.c, matrix.d, matrix.e, matrix.f);
     state.canvas->clear(backgroundColor);
     root->render(state);
-    state.canvas->rgba();
+
+    /** Changed by MTA */
```

Comment:
PlutoVG image format is ARGB32 Premultiplied not BRGA

Reply:
ARGB Premultiplied -> BGRA Non-Premultiplied

```cpp
auto width = state.canvas->width();
auto height = state.canvas->height();
auto stride = state.canvas->stride();
auto data = state.canvas->data();
for(int y = 0;y < height;y++)
{
    auto pixels = reinterpret_cast<uint32_t*>(data + stride * y);
    for(int x = 0;x < width;x++)
    {
        auto pixel = pixels[x];
        auto a = (pixel >> 24) & 0xFF;
        if(a == 0)
            continue;

        auto r = (pixel >> 16) & 0xFF;
        auto g = (pixel >> 8) & 0xFF;
        auto b = (pixel >> 0) & 0xFF;
        if(a != 255)
        {
            r = (r * 255) / a;
            g = (g * 255) / a;
            b = (b * 255) / a;
        }

        pixels[x] = (a << 24) | (r << 16) | (g << 8) | b;
    }
}
```

Reply:
aha, I see - thanks!